libcxx/include/__format/format_functions.h

```diff
@@ -91,13 +113,22 @@ class _LIBCPP_TEMPLATE_VIS __compile_time_handle {
   }
 
   template <class _Tp>
+    requires __formattable_with_constexpr<_Tp, __format_context<_CharT>>
   _LIBCPP_HIDE_FROM_ABI constexpr void __enable() {
     __parse_ = [](basic_format_parse_context<_CharT>& __ctx) {
       formatter<_Tp, _CharT> __f;
       __ctx.advance_to(__f.parse(__ctx));
```

Comment:
```suggestion
  template <class _Tp>
  _LIBCPP_HIDE_FROM_ABI constexpr void __enable() {
    static_assert(__formattable_with_constexpr<_Tp, __format_context<_CharT>>, "message");
    if constexpr (__formattable_with_constexpr<_Tp, __format_context<_CharT>>
      __parse_ = [](basic_format_parse_context<_CharT>& __ctx) {
        formatter<_Tp, _CharT> __f;
        __ctx.advance_to(__f.parse(__ctx));
      }
```

Perhaps that is also worth exploring? It's less heavyweight than the current approach. I still think the `if constexpr` goes against the goal of not reimplementing how the compiler issues diagnostics, but that could potentially be something that we get consensus on.